pkg/controller/s3/bucket/replicationConfig.go

```diff
@@ -214,3 +213,154 @@ func sortReplicationRules(rules []awss3.ReplicationRule) {
 		}
 	}
 }
+
+// LateInitialize does nothing because the resource might have been deleted by
+// the user.
+func (in *ReplicationConfigurationClient) LateInitialize(ctx context.Context, bucket *v1beta1.Bucket) error {
+	external, err := in.client.GetBucketReplicationRequest(&awss3.GetBucketReplicationInput{Bucket: awsclient.String(meta.GetExternalName(bucket))}).Send(ctx)
+	if err != nil {
+		if s3.ReplicationConfigurationNotFound(err) {
+			return nil
+		}
+		return errors.Wrap(resource.Ignore(s3.ReplicationConfigurationNotFound, err), replicationGetFailed)
+	}
+
+	if external.GetBucketReplicationOutput == nil || external.ReplicationConfiguration == nil {
+		return nil
+	}
+
+	in.logger.Debug(fmt.Sprintf("called LateInitialize for %s", reflect.TypeOf(in).Elem().Name()))
+
+	if bucket.Spec.ForProvider.ReplicationConfiguration == nil {
+		// We need the configuration to exist so we can initialize
+		bucket.Spec.ForProvider.ReplicationConfiguration = &v1beta1.ReplicationConfiguration{}
+	}
+
+	createReplicationRulesFromExternal(external.ReplicationConfiguration, bucket.Spec.ForProvider.ReplicationConfiguration)
+	return nil
+}
+
+func createReplicationRulesFromExternal(external *awss3.ReplicationConfiguration, config *v1beta1.ReplicationConfiguration) { // nolint:gocyclo
+	config.Role = awsclient.LateInitializeStringPtr(config.Role, external.Role)
+	if config.Rules == nil {
+		config.Rules = make([]v1beta1.ReplicationRule, 0)
+	}
+
+	for i, rule := range external.Rules {
```

Comment:
Same comment as other late-inited arrays. Unless we can sort them reliably with a unique identifier (could be a field or combination of fields), we can late-init only if the desired array is empty.